pkg/controller/podgc/gc_controller_test.go

```diff
@@ -194,7 +195,92 @@ func TestGCOrphaned(t *testing.T) {
 		gcc.podController = &FakeController{}
 		gcc.nodeController = &FakeController{}
 
-		gcc.gc()
+		pods, err := gcc.podStore.List(labels.Everything())
+		if err != nil {
+			t.Errorf("Error while listing all Pods: %v", err)
+			return
+		}
+		gcc.gcOrphaned(pods)
+
+		pass := true
+		for _, pod := range deletedPodNames {
+			if !test.deletedPodNames.Has(pod) {
+				pass = false
+			}
+		}
+		if len(deletedPodNames) != len(test.deletedPodNames) {
+			pass = false
+		}
+		if !pass {
+			t.Errorf("[%v]pod's deleted expected and actual did not match.\n\texpected: %v\n\tactual: %v", i, test.deletedPodNames, deletedPodNames)
+		}
+	}
+}
+
+func TestGCUnscheduledTerminating(t *testing.T) {
+	type nameToPhase struct {
+		name              string
+		phase             api.PodPhase
+		deletionTimeStamp *unversioned.Time
+		nodeName          string
+	}
+
+	testCases := []struct {
+		pods            []nameToPhase
```

Comment:
I generally find it useful to have a "name" or even better "description" per test case, which is printed with the failure, to make it easier to locate problematic test.